File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = {-1, -1};
        int start = IndexSearch(nums, target, true);
        int end = IndexSearch(nums, target, false);
        ans[0] = start;;
        ans[1] = end;
        
        return ans;
    }
    int IndexSearch(int[] nums, int target, boolean isStartIndex){
        int start = 0, end = nums.length - 1;
        int ans = -1;
        
        while (start <= end){
            int mid = start + (end - start) / 2;
            if (nums[mid] < target){
                start = mid + 1;
            }
            else if(nums[mid] > target){
                end = mid - 1;
            }
            else{
                ans = mid;
                if (isStartIndex)
                {
                    end = mid - 1;
                }
                else
                {
                    start = mid + 1;
                }
            }
        }
        return ans;
        
    }
}   
    

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        
        start = self.searchIndex(nums, target, True)
        end = self.searchIndex(nums, target, False)
        
        return [start, end] 
    
    def searchIndex(self, nums, target, leftbias):
        start,end = 0,len(nums)-1
        ans=-1
        while start<=end:
            
            mid= (start+end)//2
            if target>nums[mid]:
                start =mid+1
            elif target<nums[mid]:
                end =mid-1 
            else:
                ans =mid
                if leftbias:
                    end=mid-1 
                else:
                    start=mid+1
        return ans
                    